docs/guides/git/index.md

```diff
@@ -0,0 +1,473 @@
+
+```
+    git config --global core.autocrlf false
+```
+
+## Setting-up your public repository
+
+1. Go to [Github](http://github.com/) and create an account.
+
+2. Go to the official [Moodle Github repository](http://github.com/moodle/moodle) and click on the Fork button. You now have your own Github Moodle repository.
+
+3. Set up your SSH public key so you can push to your Github Moodle repository from your local Moodle repository. Over at Github, navigate to your Settings page and look for a section called **SSH and GPG keys**. Configuring your SSH key can be a little tricky, but following [Github's SSH setup steps](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent) will ensure the hardest part of this setup is done correctly.
+
+4. Optionally, you can set up [Github Actions](/general/development/tools/gha) to automatically test your code. This provides the advantage of catching issues without having to set up a testing environment locally and provides helpful feedback before you submit your code for review.
+
+## Setting-up your local repository
+
+Create a local clone of your Github repository by using one of these commands in your terminal:
+
+```
+    git clone git://github.com/YOUR_GITHUB_USERNAME/moodle.git YOUR_LOCALDIR
+```
+
+```
+    git clone git@github.com:YOUR_GITHUB_USERNAME/moodle.git YOUR_LOCALDIR
+```
+
+These commands do several jobs for you:
+
+- Creates a new folder
+- Initialises a new local Git repository
+- Sets your Github repository as the remote repository called 'origin'
+- Makes a local checkout of the branch 'master'
+
+:::note
+In the first example, the URL used in the command is `git://github.com`. This gives read-only access to the your Github repository, and the `git push origin` command that appears later in this document will not work.
+
+If you want to be able to update your **origin** repository, use the URL that starts with `git@github.com` (the second of the two `git clone` commands given above). This will give you read and write access to your Github repository at github.com.
+:::
+
+## Keeping your public repository up-to-date
+<!-- cspell:ignore forwardable -->
+Your fork at Github is not updated automatically. To keep it synced with the upstream Moodle repository, you have to fetch the recent changes from the official moodle.git repository. To avoid conflicts, it is strongly recommended that you never modify the standard Moodle branches. Never commit directly into **master** and **MOODLE_XXX_STABLE** branches, but instead create topic branches to work on. In Git terminology, the **master** branch and **MOODLE_XXX_STABLE** branches should always be fast-forwardable.
```

Comment:
Replace `**master**` with just backticks.

Reply:
```suggestion
Your fork at Github is not updated automatically. To keep it synced with the upstream Moodle repository, you have to fetch the recent changes from the official moodle.git repository. To avoid conflicts, it is strongly recommended that you never modify the standard Moodle branches. Never commit directly into `master` and `MOODLE_XXX_STABLE` branches, but instead create topic branches to work on. In Git terminology, the `master` branch and `MOODLE_XXX_STABLE` branches should always be fast-forwardable.
```